src/Cryptol/Symbolic/SBV.hs

```diff
@@ -250,6 +279,25 @@ runMultiProvers pc lPutStrLn provers callSolver processResult e = do
               do forM_ others (\a -> X.throwTo (asyncThreadId a) ExitSuccess)
                  return r
 
+-- | Wrap a solver call with a given timeout. If the timeout is reached, then
+--   the default result is returned and the solver call is terminated.
```

Comment:
As I understand it, this code is only necessary to work around an issue in the way SBV interacts with Yices. Is that correct? If so, worth stating as much in the Haddocks.

Reply:
It is, but at the moment it's actually wrapping all solver calls regardless, it just happens to be strictly necessary with Yices.

In general there's no harm doing this, it's just an additional measure to ensure that the timeout is respected even if the solver misbehaves somehow.

I can note all of this in the Haddocks.